lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,293 @@
+              routingType: dependency as RoutingType,
+            });
+          }
+        }
+      }
+    }
+    return ChainConfigManager._chainConfigsWithDependencies;
+  }
+
+  static get reverseConfigs(): { [routeType: string]: number[] } {
+    if (ChainConfigManager._reverseConfigs) {
+      return ChainConfigManager._reverseConfigs;
+    }
+    ChainConfigManager._reverseConfigs = {};
+    for (const chainId in ChainConfigManager.chainConfigsWithDependencies) {
+      for (const supportedRoutingType of ChainConfigManager.chainConfigsWithDependencies[chainId].routingTypes) {
+        if (!ChainConfigManager._reverseConfigs[supportedRoutingType.routingType]) {
+          ChainConfigManager._reverseConfigs[supportedRoutingType.routingType] = [];
+        }
+        ChainConfigManager._reverseConfigs[supportedRoutingType.routingType].push(parseInt(chainId));
+      }
+    }
+    return ChainConfigManager._reverseConfigs;
+  }
+
+  /**
+   * @returns all ChainIds
+   */
+  public static getChainIds(): ChainId[] {
+    return Object.keys(ChainConfigManager.chainConfigsWithDependencies).map((c) => Number.parseInt(c));
+  }
+
+  /**
+   * @param routingType The RoutingType to check
+   * @returns all chains that support given RoutingType
+   */
+  public static getChainIdsByRoutingType(routingType: RoutingType): ChainId[] {
+    return ChainConfigManager.reverseConfigs[routingType] || [];
+  }
+
+  /**
+   * @param alarmEnabled Alarms set or not
+   * @returns all chains that have the given alarm setting
+   */
+  public static getChainIdsByAlarmSetting(alarmEnabled: boolean): ChainId[] {
+    const chainIds: ChainId[] = [];
+    for (const chainId in ChainConfigManager.chainConfigsWithDependencies) {
+      if (ChainConfigManager.chainConfigsWithDependencies[chainId].alarmEnabled == alarmEnabled) {
+        chainIds.push(Number.parseInt(chainId));
+      }
+    }
+    return chainIds;
+  }
+
+  /**
+   * @param chainId the ChainId to check
+   * @param routingType the RoutingType to check
+   * @returns true when ChainId supports provided RoutingType
+   */
+  public static chainSupportsRoutingType(chainId: ChainId, routingType: RoutingType) {
+    return (
+      chainId in ChainConfigManager.chainConfigsWithDependencies &&
+      ChainConfigManager.chainConfigsWithDependencies[chainId].routingTypes.some((r) => r.routingType == routingType)
+    );
+  }
+
+  /**
+   * @param chainId the ChainId to check
+   * @param routingType the RoutingType to check
+   * @returns the QuoteConfig for the provided ChainId and RoutingType
+   */
+  public static getQuoteConfig(chainId: ChainId, routingType: RoutingType) {
+    if (!(chainId in ChainConfigManager.chainConfigsWithDependencies)) {
+      throw new Error(`Unexpected chainId ${chainId}`);
+    }
+    // Should only return one element if exists
+    const quoteConfig = ChainConfigManager.chainConfigsWithDependencies[chainId].routingTypes.filter(
```

Comment:
if `chainConfigsWithDependencies` ensures that each routingType is only specified once in the object I think I would prefer something like .find() instead of filter, or some check that there are not multiple quoteConfigs returned

Reply:
I like it!